pages/docs/tutorials/native/using-intellij-idea.md

```diff
@@ -1,50 +1,145 @@
+   ![Configure a native application]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-2.png') }})
 
-![Wizard Step Two]({{ url_for('tutorial_img', filename='native/using-intellij-idea/wizard-2.png')}})
+Your project opens. By default, the wizard creates the necessary `main.kt` file with the code that prints "Hello, Kotlin/Native!" to the standard output.
+The `build.gradle.kts` file contains information about target platforms (Windows, Linux, macOS).
 
-On clicking **Next**, enter the path and name for the project.
+## Run the application
 
-![Wizard Step Three]({{ url_for('tutorial_img', filename='native/using-intellij-idea/wizard-3.png')}})
+Start the application by clicking **Run** next to the run configuration at the top of the screen.
 
-This will complete the process and open the newly created project in the IDE. By default the wizard will create the necessary
-`Sample<TARGET>.kt` file and provide code for writing some string to the standard output. Note that `<TARGET>` varies based on the operating
-system the project was created on (Windows, Linux, macOS).
+![Run the application]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-3.png') }})
 
-![Project]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-1.png')}})
+IntelliJ IDEA opens the **Run** tab and shows the output:
+![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-4.png') }})
 
-To run the project, simply invoke the [Run command in the IDE](https://www.jetbrains.com/help/idea/running-applications.html), by using the corresponding shortcuts or from the Run menu
+## Update the application
 
-![Run]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-2.png')}})
+### Count characters in your name
+
+1. Open the file `main.kt` in `src/<your_app_name>Main/kotlin`.
+
+   The `src` directory contains Kotlin source files and resources. The file `main.kt` includes sample code that prints "Hello, Kotlin/Native!" using the [prinln](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/println.html) function.
+
+2. Change the code of `main()`:
+
+   * Use the [readLine](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/read-line.html) function to read the input value to the `name` variable.
+   * Check the input value with the [safe call operator](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls) `?.`. 
+   * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) to insert your name into the string by adding a dollar sign `$` and enclosing it in curly braces – `${name?.length}`. 
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+   
+   ```kotlin
+   fun main() {
+       println("Hello, enter your name:")
+   
+       var name = readLine()
+       println("Your name contains ${name?.length} letters")
```

Comment:
I think it's a bit strange in case of null. `null letters`? May be provide readable error with `error()` function?